### PR TITLE
Add endpoints for managing band changes

### PR DESCRIPTION
**Pull request by pixeltrix**

### CODE REVIEW

**Comment by pixeltrix** [comment]
These are required for the first story about showing projections - rest of band change APIs to be added in a separate PR.

File: BonusCalcApi.Tests/V1/Controllers/BandChangesControllerTests.cs [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+                .ThrowsAsync(new ResourceNotFoundException("Open bonus period not found"));
+
+            // Act
+            var objectResult = await _classUnderTest.GetProjectedChanges();
+            var statusCode = GetStatusCode(objectResult);
+            var result = GetResultData<ProblemDetails>(objectResult);
+
+            // Assert
+            statusCode.Should().Be((int) HttpStatusCode.NotFound);
+            result.Status.Should().Be((int) HttpStatusCode.NotFound);
```
Comment (Asura5) [inline BonusCalcApi.Tests/V1/Controllers/BandChangesControllerTests.cs:118]: Are these not effectively testing the same thing?

File: BonusCalcApi/V1/Controllers/BandChangesController.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace BonusCalcApi.V1.Controllers
+{
+    [ApiController]
+    [Route("api/v1/changes")]
```
Comment (Asura5) [inline BonusCalcApi/V1/Controllers/BandChangesController.cs:15]: It's a minor gripe but "changes" could mean changes to anything here - it's the first post-version element in the route. Can we use /bandchanges or /bands/changes

Comment (pixeltrix) [reply BonusCalcApi/V1/Controllers/BandChangesController.cs:15]: How about `/band-changes` ?

**Pull request merged**